src/contexts/products-context.js

```diff
@@ -0,0 +1,82 @@
+import axios from "axios";
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { cartReducer } from "../reducers/cart-reducer";
+export const ProductsContext = createContext();
+
+export function ProductsProvider({ children }) {
+  const [productsData, setProductsData] = useState([]);
```

Comment:
yes shubham actually i used the previously created code for the FE only and integrated it......although the reason is that i am sort of creating the copy of the incoming data so that the sorting and filtering can be done directly.